src/flamenco/vm/fd_vm_interp_core.c

```diff
@@ -682,22 +693,19 @@
           is a best-effort implementation that should match the VM state
           in all ways except error code. */
 
-      /* Note the original implementation had the imm magic number
-          check after the calldest check.  But fd_pchash_inverse of the
-          magic number is 0xb00c380U.  This is beyond possible text_cnt
-          values.  So we do it first to simplify the code and clean up
-          fault handling. */
-
-      if( FD_UNLIKELY( imm==0x71e3cf81U ) ){
+      /* Special case to handle entrypoint.
+         ebpf::hash_symbol_name(b"entrypoint") = 0xb00c380, and
+         fd_pchash_inverse( 0xb00c380U ) = 0x71e3cf81U */
+      if( FD_UNLIKELY( imm==0x71e3cf81U ) ) {
         FD_VM_INTERP_STACK_PUSH;
-        pc = entry_pc; /* FIXME: MAGIC NUMBER */
+        pc = entry_pc - 1;
```

Comment:
Interesting change...

Reply:
note the old `pc--` on line 712. I found it a bit less readable, so I moved the -1 inside the branches.